src/Api/Api.js

```diff
@@ -54,6 +55,35 @@ export default class Api {
 		this.deviceIdentification = deviceIdentification;
 	}
 
+	setCustomHeaders(customHeaders) {
+		// Ignore empty headers
+		if(!customHeaders)
+			return;
+
+		ow(customHeaders, "customHeaders", ow.object);
+
+		Object.keys(customHeaders).forEach((customHeader) => {
+			const lowerCaseCustomHeader = customHeader.toLowerCase();
+
+			ow(lowerCaseCustomHeader, customHeader, ow.string.nonEmpty);
+
+			// Headers must start with a `x-` prefix
+			ow(lowerCaseCustomHeader, customHeader, ow.string.startsWith("x-"));
+
+			// Headers must not start with OUR prefix
+			ow(lowerCaseCustomHeader, customHeader, ow.string.not.startsWith("x-parley-"));
+			ow(lowerCaseCustomHeader, customHeader, ow.string.not.startsWith("x-iris-"));
```

Comment:
Hier heb je nu geen test voor geschreven of zie ik dat verkeerd?

Reply:
Klopt, nu toegevoegd